apache/metron
# METRON-2085: [UI] Alerts UI Details Pane: naming meta alerts is broken

## Contributor Comments

Link to the original ASF Jira: https://issues.apache.org/jira/browse/METRON-2085

1. When you click on the name of the meta alert in the detail pane, the input field to change it appears but there's nothing inside but he current meta alert name should be there (because you're changing that). After applying this patch, the user can see the current meta alert name in the input field. 

2. It's impossible to click on the cancel button when it comes to editing the name of the meta alert because an invisible html element covers it (the html element that wraps the close button of the entire pane). After applying this patch the user will be able to click on the cancel button.

BONUS: I added some keyboard functionalities which means if the user presses enter after updating the meta alert name, it's going to persist the change i.e. save the new name. When the user presses escape button while editing, it means a "cancel".

Test:

1. Go to the meta alerts list screen
2. Click on an existing meta alert row in order to open the details pane.
3. If you don't have any meta alerts in the list, create one because it has to be a meta alert if you want to add a different name. You cannot rename a regular alert.
4. In the opened pane, at the top, you can click on the current name of the meta alert. It turns into an editable input field where you can change the name.
5. It's quite straightforward to test the functionality based on the summary of the problem in the beginning of the description. Try to rename it and save it by hitting enter or clicking on the check icon on the right. Try to rename it and cancel the process by hitting the escape button or clicking on the cross icon next to the check icon on the right.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
